File: app/src/main/java/com/example/carrent/LoginActivity.java
package com.example.carrent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class LoginActivity extends AppCompatActivity {


    private Button mBtregister;
    private Button mBtlogin;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        mBtlogin = (Button) findViewById(R.id.bt_login);
        mBtlogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getBaseContext(),MainActivity.class));
            }
        });
        mBtregister = (Button) findViewById(R.id.bt_register1);
        mBtregister.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(getBaseContext(), RegisterActivity.class));
            }
        });

    }
}

File: app/src/main/java/com/example/carrent/LoginActivity.kt
package com.example.carrent

import android.content.Intent
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : BaseActivity() {

    private lateinit var viewModel: LoginViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        viewModel = ViewModelProvider(this).get(LoginViewModel::class.java)

        bt_login.setOnClickListener {
            val username = edt_loginusername.getContents()
            val password = edt_loginpassword.getContents()
            viewModel.requestLogin(username, password)
        }

        bt_register1.setOnClickListener {
            startActivity(Intent(baseContext, RegisterActivity::class.java))
        }

        viewModel.responselogin().observe {
            if (it) {
                startActivity(Intent(baseContext, MainActivity::class.java))
                // go to main
            } else {
                toast("ชื่อหรือรหัสผ่านไม่ถูกต้อง")
            }
        }

        viewModel.responseToast().observe {
            toast(it)
        }
    }
}
